src/Quartz/CronExpression.cs

```diff
@@ -1681,265 +1679,289 @@ private SortedSet<int> CalculateDaysOfMonth(DateTimeOffset dt)
             return results;
         }
 
-        /// <summary>
-        /// Progress next fire time day
-        /// </summary>
-        /// <param name="d">NextFireTimeCheck</param>
-        private NextFireTimeCursor ProgressNextFireTimeDay(DateTimeOffset d)
+        private NextFireTimeCursor ProgressNextFireTimeDayOfMonth(DateTimeOffset d)
         {
             var day = d.Day;
             var mon = d.Month;
             var t = -1;
             var tmon = mon;
 
-            // get day
-            var dayOfMSpec = !daysOfMonth.Contains(CronExpressionConstants.NoSpec);
-            var dayOfWSpec = !daysOfWeek.Contains(CronExpressionConstants.NoSpec);
-            SortedSet<int> tailDays;
-            if (dayOfMSpec && !dayOfWSpec)
+            // get day by day of month rule
+            var daysOfMonthCalculated = CalculateDaysOfMonth(d);
+            var tailDays = daysOfMonthCalculated.TailSet(d.Day);
+            var found = tailDays.Any();
```

Comment:
should avoid LINQ features, `.Count` is always faster, probably was old code

Reply:
fixed.